docs/ARCHITECTURE.md

```diff
@@ -83,30 +83,42 @@ Sources for the udev daemon and command-line tool (single binary) can be found u
 Source files found under `src/test/` implement unit-level testing, mostly for
 modules found in `src/basic/` and `src/shared/`, but not exclusively. Each test
 file is compiled in a standalone binary that can be run to exercise the
-corresponding module. While most of the tests can be ran by any user, some
+corresponding module. While most of the tests can be run by any user, some
 require privileges, and will attempt to clearly log about what they need
 (mostly in the form of effective capabilities). These tests are self-contained,
 and generally safe to run on the host without side effects.
 
 Ideally, every module in `src/basic/` and `src/shared/` should have a
 corresponding unit test under `src/test/`, exercising every helper function.
 
+## Fuzzer test cases
+
+Files under `test/fuzz/` contain input data for fuzzers, one subdirectory for
+each fuzzer. Some of the files are "seed corpora", i.e. files that contain
+lists of settings and input values intended to generate initial coverage, and
+other files are samples saved by the fuzzing engines when they find an issue.
+
+When adding new input samples under `test/fuzz/*/`, please use some
+short-but-meaningful name. Names of meson tests include the input file name and
```

Comment:
Those long names come from OSS-Fuzz usually. I think they are supposed to be renamed to `oss-fuzz-...` to make it easier to find issues they've triggered. If it could be officially documented it would be great.

Reply:
*I* used the naming of `oss-fuzz-*`, but ultimately it's not that useful. The link to the issue should be part of the commit description, so it's easy to click and so on. But for the name, I think `crash-zero-length` is much more useful than  `crash-oss-fuzz-12345`.

Reply:
> But for the name, I think crash-zero-length is much more useful than crash-oss-fuzz-12345.

Can't argue with that. I'm not sure how easy it is to come up with descriptive names like that in general but I agree.

Reply:
I think "short-but-meaningful" covers both approaches. I think that should be "good enough": if people have an idea for a good name, that can use that. Otherwise, `oss-fuzz-nnnn` is also acceptable.